packages/client-sdk-nodejs/src/internal/internal-auth-client.ts

```diff
@@ -146,20 +167,83 @@ export class InternalAuthClient implements IAuthClient {
       );
     });
   }
+
+  public async generateDisposableToken(
+    scope: TemporaryTokenScope,
+    expiresIn: ExpiresIn
+  ): Promise<GenerateDisposableToken.Response> {
+    const tokenClient = new token.token.TokenClient(
+      this.creds.getTokenEndpoint(),
+      ChannelCredentials.createSsl()
+    );
+
+    try {
+      validateDisposableTokenExpiry(expiresIn);
+    } catch (err) {
+      return new GenerateDisposableToken.Error(normalizeSdkError(err as Error));
+    }
+    const expires = new Expires({
+      valid_for_seconds: expiresIn.seconds(),
+    });
+
+    let permissions;
+    try {
+      permissions = permissionsFromScope(scope);
+    } catch (err) {
+      return new GenerateDisposableToken.Error(normalizeSdkError(err as Error));
+    }
+
+    const request = token.token._GenerateDisposableTokenRequest.fromObject({
+      expires: expires,
+      auth_token: this.creds.getAuthToken(),
+      permissions: permissions,
+    });
+
+    return await new Promise<GenerateDisposableToken.Response>(resolve => {
+      tokenClient.GenerateDisposableToken(
+        request,
+        {interceptors: this.interceptors},
+        (err, resp) => {
+          if (err || !resp) {
+            resolve(
+              new GenerateDisposableToken.Error(cacheServiceErrorMapper(err))
+            );
+          } else {
+            resolve(
+              new GenerateDisposableToken.Success(
+                resp.api_key,
+                resp.endpoint,
+                ExpiresAt.fromEpoch(resp.valid_until)
+              )
+            );
+          }
+        }
+      );
+    });
+  }
 }
 
 export function permissionsFromScope(
-  scope: TokenScope
-): grpcAuth._GenerateApiTokenRequest.Permissions {
-  const result = new grpcAuth._GenerateApiTokenRequest.Permissions();
+  scope: TokenScope | TemporaryTokenScope
+): permission_messages.Permissions {
+  const result = new permission_messages.Permissions();
   if (scope instanceof InternalSuperUserPermissions) {
-    result.super_user =
-      grpcAuth._GenerateApiTokenRequest.SuperUserPermissions.SuperUser;
+    result.super_user = permission_messages.SuperUserPermissions.SuperUser;
+    return result;
+  } else if (
+    !(scope instanceof PredefinedScope) &&
```

Comment:
To prevent this error:

> `Argument of type 'Permissions | PredefinedScope | DisposableTokenCachePermissions' is not assignable to parameter of type 'DisposableTokenScope'.
  Type 'PredefinedScope' is not assignable to type 'DisposableTokenScope'.ts(2345)`